includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -707,30 +707,29 @@ public function get_charge( $charge_id ) {
 	/**
 	 * List disputes
 	 *
+	 * @param  int $page The page index to retrieve.
+	 * @param  int $page_size The number of items the page contains.
 	 * @return array
 	 * @throws API_Exception - Exception thrown on request failure.
 	 */
-	public function list_disputes() {
+	public function list_disputes( int $page = 0, int $page_size = 25 ):array {
 		$query = [
-			'limit' => 100,
+			'limit'    => 100,
+			'page'     => $page,
+			'pagesize' => $page_size,
 		];
 
-		$disputes = $this->request( $query, self::DISPUTES_API, self::GET );
-
-		// Add WooCommerce order information to each dispute.
-		if ( isset( $disputes['data'] ) ) {
-			foreach ( $disputes['data'] as &$dispute ) {
-				try {
-					// Wrap with try/catch to avoid failing whole request because of a single dispute.
-					$dispute = $this->add_order_info_to_object( $dispute['charge']['id'], $dispute );
-				} catch ( Exception $e ) {
-					// TODO: Log the error once Logger PR (#326) is merged.
-					continue;
-				}
-			}
-		}
```

Comment:
Why is this block removed? We need this to have a link to order in the UI.

Reply:
I think it would be more efficient to request dispute data with expanded `charge` object in the back-end, and extract order number from the metadata. `add_order_info_to_object` can be a costly operation.

Reply:
I see a possibility to improve performance but we can't simply drop this block and add charge in the back-end. Yes, we could remove the query to get the order number but we still need mapping to fetch full order and build the info array that `add_order_info_to_object` does. Back-end doesn't have that information available (nor should it have the responsibility of doing that).

I would also argue the overall performance gain when you take into account necessary back-end changes.

Reply:
@KarlisJ it was removed by mistake, bringing it back.

Reply:
This block was removed by a mistake, bringing it back.